### PR TITLE
Link to IO.string.format and Regexp from string docs

### PR DESCRIPTION
**Pull request by mppf**
It seems to me it would be useful to note the existence of these other routines.
In particular, I remember somebody asking if it's possible to go from an
integer storing an ASCII character code to a 1-character string. The answer
is yes, the format routine can do that. But that might be difficult to discover
with the current version of the documentation.

Whether format is an *efficient* way to do this is another question.

Reviewed by @benharsh - thanks!

### CODE REVIEW

**Comment by mppf** [comment]
@benharsh are you interested in reviewing?

File: modules/internal/String.chpl [hunk 1]
```diff
@@ -41,10 +41,16 @@ private module BaseStringType {
 // It might be worth moving them here for documentation - KB Feb 2016
 
 /*
-  The following documentation consists of functions and methods used to
+  The following documentation shows functions and methods used to
   manipulate and process Chapel strings. See the :mod:`IO` documentation for
   more information on reading, writing, and formatting strings.
```
Comment (benharsh) [inline modules/internal/String.chpl]: I like these changes. Should this sentence about the IO documentation be moved or removed?

Comment (mppf) [inline modules/internal/String.chpl]: I think I've adjusted it appropriately - have another look?

**Comment by benharsh** [comment]
:+1:

**Pull request merged**